### PR TITLE
[breaking][libra framework] Add tests to increase testing coverage of the stdlib

### PR DESCRIPTION
**Pull request by tzakian**
This PR adds a bunch of tests to increase the coverage of the Move stdlib. (see below for a per-module summary). 

There were a couple changes to the modules, but they were kept as minimal as possible for this PR. There will need to be a follow up PR to update validatorconfig/libraSystem logic to perform role checks on accounts. But, this is a breaking change so going to wait on these to roll in to another PR. 

A future PR will add a land-blocking test to make sure these coverage numbers do not regress.

```
+-------------------------+
| Move Coverage Summary   |
+-------------------------+
Module 00000000000000000000000000000001::Debug
>>> % Module coverage: NaN
Module 00000000000000000000000000000001::Hash
>>> % Module coverage: NaN
Module 00000000000000000000000000000001::Authenticator
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::ChainId
>>> % Module coverage: 72.41
Module 00000000000000000000000000000001::LibraConfig
>>> % Module coverage: 94.65
Module 00000000000000000000000000000001::RegisteredCurrencies
>>> % Module coverage: 91.30
Module 00000000000000000000000000000001::FixedPoint32
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::Libra
>>> % Module coverage: 97.58
Module 00000000000000000000000000000001::Coin1
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::Coin2
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::Compare
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::Signer
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::DesignatedDealer
>>> % Module coverage: 99.00
Module 00000000000000000000000000000001::VASP
>>> % Module coverage: 96.44
Module 00000000000000000000000000000001::Signature
>>> % Module coverage: NaN
Module 00000000000000000000000000000001::LBR
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::DualAttestation
>>> % Module coverage: 96.75
Module 00000000000000000000000000000001::TransactionFee
>>> % Module coverage: 90.98
Module 00000000000000000000000000000001::ValidatorOperatorConfig
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::Option
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::ValidatorConfig
>>> % Module coverage: 98.51
Module 00000000000000000000000000000001::SlidingNonce
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::CoreAddresses
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::LibraTransactionTimeout
>>> % Module coverage: 93.85
Module 00000000000000000000000000000001::LibraAccount
>>> % Module coverage: 98.94
Module 00000000000000000000000000000001::LibraWriteSetManager
>>> % Module coverage: 94.94
Module 00000000000000000000000000000001::LibraVersion
>>> % Module coverage: 90.70
Module 00000000000000000000000000000001::LibraVMConfig
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::LibraSystem
>>> % Module coverage: 97.17
Module 00000000000000000000000000000001::LibraBlock
>>> % Module coverage: 95.51
Module 00000000000000000000000000000001::Genesis
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::Offer
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::RecoveryAddress
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::LibraTimestamp
>>> % Module coverage: 94.89
Module 00000000000000000000000000000001::SharedEd25519PublicKey
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::Roles
>>> % Module coverage: 94.79
Module 00000000000000000000000000000001::Vector
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::LCS
>>> % Module coverage: NaN
Module 00000000000000000000000000000001::Event
>>> % Module coverage: 100.00
Module 00000000000000000000000000000001::AccountFreezing
>>> % Module coverage: 96.43
Module 00000000000000000000000000000001::AccountLimits
>>> % Module coverage: 99.47
+-------------------------+
| % Move Coverage: 97.70  |
+-------------------------+
```

### CODE REVIEW

File: language/stdlib/modules/VASP.move [hunk 1]
```diff
@@ -52,11 +52,13 @@ module VASP {
     public fun publish_parent_vasp_credential(vasp: &signer, lr_account: &signer) {
         assert(Roles::has_libra_root_role(lr_account), ENOT_LIBRA_ROOT);
         let vasp_addr = Signer::address_of(vasp);
-        assert(!is_vasp(vasp_addr), ENOT_A_VASP);
+        assert(Roles::has_parent_VASP_role(vasp), ENOT_A_PARENT_VASP);
```
Comment (tzakian) [inline language/stdlib/modules/VASP.move:55]: @shazqadeer This assertion is what I was mentioning this morning  that should allow us to prove "has parent vasp resource ==> has parent vasp role"

File: language/stdlib/modules/LibraAccount.move [hunk 2]
```diff
@@ -887,8 +887,8 @@ module LibraAccount {
         auth_key_prefix: vector<u8>,
         human_name: vector<u8>,
     ) {
-        assert(Roles::has_libra_root_role(creator_account), ENOT_LIBRA_ROOT);
```
Comment (sblackshear) [inline language/stdlib/modules/LibraAccount.move:890]: This is checked in `Roles::new_validator_role`?

File: language/stdlib/modules/LibraTimestamp.move [hunk 3]
```diff
@@ -47,6 +48,7 @@ module LibraTimestamp {
             exists<CurrentTimeMicroseconds>(CoreAddresses::LIBRA_ROOT_ADDRESS()) && now_microseconds() == 0,
             ETIME_NOT_INITIALIZED
         );
+        assert(!exists<TimeHasStarted>(Signer::address_of(lr_account)), ETIME_ALREADY_STARTED);
```
Comment (sblackshear) [inline language/stdlib/modules/LibraTimestamp.move:51]: Could we use `EGENESIS_ONLY` (or equivalent) here, or is this a different problem?

File: language/stdlib/modules/LibraAccount.move (hunk 2 above)
Comment (tzakian) [reply language/stdlib/modules/LibraAccount.move:890]: yep! Good catch. This is checked in roles (removed to be more in-line with the way checks are performed in the other `create_account` functions). I'll add a comment.

File: language/stdlib/modules/LibraTimestamp.move (hunk 3 above)
Comment (tzakian) [reply language/stdlib/modules/LibraTimestamp.move:51]: I like EGENESIS_ONLY, and it makes sense here. I'll update.

**Pull request closed without merging**